nixos/modules/services/web-apps/comfyui.nix

```diff
@@ -0,0 +1,498 @@
+      };
+
+      max-upload-size = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+      };
+
+      multi-user = mkOption {
+        type = types.bool;
+        default = false;
+      };
+
+      cuda-device = mkOption {
+        type = types.nullOr types.str;
+        description = ''The CUDA device to use.  Query for this using lspci or lshw.  Leave as null to auto-detect and/or use Nix CUDA settings (verify this before merging!).'';
+        default = null;
+      };
+
+      verbose = mkOption {
+        type = types.bool;
+        description = ''Use verbose logging.'';
+        default = false;
+      };
+
+      cross-attention = mkOption {
+        # TODO: Learn2enum in Nix.
+        # Valid types should be "split", "quad", and "pytorch".
+        type = types.nullOr types.enum;
+        # TODO: Learn what cross-attention is, and describe it here.
+        description = '' '';
+        default = null;
+      };
+
+      preview-method = mkOption {
+        type = types.nullOr types.oneOf [
+          "none"
+          "auto"
+          "latent2rgb"
+          "taesd"
+        ];
+        description = '' '';
+        default = null;
+      };
+
+      # Argument dump:
+      #  usage: comfyui [-h] [--listen [IP]] [--port PORT]
+      #                 [--enable-cors-header [ORIGIN]]
+      #                 [--max-upload-size MAX_UPLOAD_SIZE]
+      #                 [--extra-model-paths-config PATH [PATH ...]]
+      #                 [--output-directory OUTPUT_DIRECTORY]
+      #                 [--temp-directory TEMP_DIRECTORY]
+      #                 [--input-directory INPUT_DIRECTORY] [--auto-launch]
+      #                 [--disable-auto-launch] [--cuda-device DEVICE_ID]
+      #                 [--cuda-malloc | --disable-cuda-malloc]
+      #                 [--dont-upcast-attention] [--force-fp32 | --force-fp16]
+      #                 [--bf16-unet | --fp16-unet | --fp8_e4m3fn-unet | --fp8_e5m2-unet]
+      #                 [--fp16-vae | --fp32-vae | --bf16-vae] [--cpu-vae]
+      #                 [--fp8_e4m3fn-text-enc | --fp8_e5m2-text-enc | --fp16-text-enc | --fp32-text-enc]
+      #                 [--directml [DIRECTML_DEVICE]] [--disable-ipex-optimize]
+      #                 [--preview-method [none,auto,latent2rgb,taesd]]
+      #                 [--use-split-cross-attention | --use-quad-cross-attention | --use-pytorch-cross-attention]
+      #                 [--disable-xformers]
+      #                 [--gpu-only | --highvram | --normalvram | --lowvram | --novram | --cpu]
+      #                 [--disable-smart-memory] [--deterministic]
+      #                 [--dont-print-server] [--quick-test-for-ci]
+      #                 [--windows-standalone-build] [--disable-metadata]
+      #                 [--multi-user] [--verbose]
+      extraArgs = mkOption {
+        type = types.attrsOf types.str;
+        default = {};
+        example = "--preview-method auto";
```

Comment:
This example has type `str` but the option type is `attrsOf str`

Reply:
Good catch.  Corrected!